lib/dictBuilder/zdict.c

```diff
@@ -810,6 +810,31 @@ static size_t ZDICT_fastSampling(void* dictBuffer, size_t dictSize,
     return nbSegments * DIB_FASTSEGMENTSIZE;
 }
 
+size_t ZDICT_addEntropyTablesFromBuffer_advanced(void* dictBuffer, size_t dictContentSize, size_t dictBufferCapacity,
+                                                 const void* samplesBuffer, const size_t* samplesSizes, unsigned nbSamples,
+                                                 ZDICT_params_t params)
+{
+    unsigned compressionLevel = params.compressionLevel;
+    if (compressionLevel==0) compressionLevel = g_compressionLevel_default;
+
+    size_t hSize;
```

Comment:
Unfortunately, we only use variable declaration at beginning of scope, for maximum portability.
You can either declare `hSize` higher in the block,
or change `compressionLevel` declaration to become a one-liner : 

`unsigned const compressionLevel = params.compressionLevel ? params.compressionLevel : g_compressionLevel_default; `

which I would favor to grab the `const` property. 
But that's just a coding taste...